From 4f13e68b1a58d5266ba9aaff785c0b152e9bbd79 Mon Sep 17 00:00:00 2001
From: Tan Jia Yong <jia.yong.tan@intel.com>
Date: Thu, 24 Oct 2024 14:54:47 +0800
Subject: [PATCH] Fix loadkeys shell process selinux permission denied

The shell process running need to search the root directory
with tmpfs_t context

Audit denied logs:
avc:  denied  { search } for  pid=780 comm="sh" name="/"
dev="overlay" ino=2 scontext=system_u:system_r:loadkeys_t:s0
tcontext=system_u:object_r:tmpfs_t:s0 tclass=dir permissive=1

Signed-off-by: Tan Jia Yong <jia.yong.tan@intel.com>
---
 policy/modules/apps/loadkeys.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/apps/loadkeys.te b/policy/modules/apps/loadkeys.te
index 2263e9dba..1b18c4415 100644
--- a/policy/modules/apps/loadkeys.te
+++ b/policy/modules/apps/loadkeys.te
@@ -37,6 +37,7 @@ files_search_src(loadkeys_t)
 files_search_tmp(loadkeys_t)
 files_dontaudit_getattr_all_dirs(loadkeys_t)
 files_read_var_symlinks(loadkeys_t)
+fs_search_tmpfs(loadkeys_t)
 
 term_dontaudit_use_console(loadkeys_t)
 term_use_unallocated_ttys(loadkeys_t)
-- 
2.37.3

